frontend/src/components/RegistrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { ChangeEvent, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import InputField from '../InputField/InputField';
+import Button from '../Button/Button';
+import { UseSignUp } from '../../hooks/useSignUp';
+import { UserCredentialTypes } from '../../types/UserTypes';
+
+const accountFields: UserCredentialTypes = {
+    email: '',
+    password: '',
+    name: '',
+    surname: '',
+};
+
+function RegistrationForm() {
+    const [inputFields, setInputFields] =
+        useState<UserCredentialTypes>(accountFields);
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    const navigate = useNavigate();
+    const [isSigning, setIsSignup] = useState(false);
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        const { name: fieldName, value: fieldValue } = event.target;
+        console.log(`Fieldname: ${fieldName} FieldValue: ${fieldValue}`);
+        setInputFields({ ...inputFields, [fieldName]: fieldValue });
+    };
+
+    const handleSignUp = async () => {
+        const user = await UseSignUp({ ...inputFields });
+    };
+
+    return (
+        <section className="registration">
+            <form className="registration__form" onSubmit={handleSignUp}>
```

Comment:
Try to use React forms lib, i know that you will have to delete half of your code but it's must have lib to learn

Reply:
Which ones?